equinox/nn/_conv.py

```diff
@@ -393,27 +466,72 @@ def __init__(
         else:
             self.bias = None
 
+        padding = _padding_init(padding, num_spatial_dims)
+        padding_mode = _padding_mode_init(padding_mode)
+        if padding_mode in ("REFLECT", "REPLICATE"):
+            raise NotImplementedError(
+                "'REFLECT' or 'REPLICATE' padding mode is not implemented"
+            )
+        elif padding_mode == "CIRCULAR":
+            if np.any(output_padding):
```

Comment:
Nit: I prefer avoiding implicit-cast-to-bool. (Here for the int->bool conversion on each element.) Can this be `if any(x !=0 for x in output_padding)`?